js_modules/dagster-ui/packages/ui-core/src/ui/Filters/doesFilterArrayMatchValueArray.tsx

```diff
@@ -0,0 +1,25 @@
+import isEqual from 'lodash/isEqual';
+
+// This function checks if every element in `filterArray` has at least one match in `valueArray` based on the provided `isMatch` comparison function.
+// - `filterArray`: The array containing elements to be matched.
+// - `valueArray`: The array to search for matches.
+// - `isMatch`: A custom comparator function (defaults to deep equality using `lodash/isEqual`).
+// Returns `false` if `filterArray` is non-empty and `valueArray` is empty (no matches possible).
+// Otherwise, checks if all elements in `filterArray` have at least one corresponding match in `valueArray`.
+// Uses `Array.prototype.some()` to verify if any `filterArray` element lacks a match and returns `false` in such cases.
+export function doesFilterArrayMatchValueArray<T, V>(
```

Comment:
Moved this to a separate file so that the webworker doesnt end up pulling React.